lib/mongo/crypt/data_key_context.rb

```diff
@@ -25,18 +25,104 @@ class DataKeyContext < Context
       #
       # @param [ Mongo::Crypt::Handle ] mongocrypt a Handle that
       #   wraps a mongocrypt_t object used to create a new mongocrypt_ctx_t
-      def initialize(mongocrypt)
-        # This initializer will eventually take more arguments:
-        # - kms_providers (just supporting local for right now)
-        # - options: master key (only relevant to AWS) and key_alt_names (not required for POC)
+      # @param [ Mongo::Crypt::EncryptionIO ] io An object that performs all
+      #   driver I/O on behalf of libmongocrypt
+      # @param [ String ] kms_provider The KMS provider to use. Options are
+      #   "aws" and "local".
+      # @param [ Hash ] options Data key creation options.
+      #
+      # @option [ Hash ] :masterkey A Hash of options related to the AWS
+      #   KMS provider option. Required if kms_provider is "aws".
+      #   - :region [ String ] The The AWS region of the master key (required).
+      #   - :key [ String ] The Amazon Resource Name (ARN) of the master key (required).
+      #   - :endpoint [ String ] An alternate host to send KMS requests to (optional).
+      def initialize(mongocrypt, io, kms_provider, options={})
+        unless ['aws', 'local'].include?(kms_provider)
+          raise ArgumentError.new(
+            "#{kms_provider} is an invalid kms provider. " +
+            "Valid options are 'aws' and 'local'"
+          )
+        end
+
+        super(mongocrypt, io)
+
+        Binding.ctx_setopt_masterkey_local(self) if kms_provider == 'local'
```

Comment:
Can this and the following if statement be organized into a single statement, either case/when or if/elsif please.